### PR TITLE
fix(test): Support running connection tests with proxy set 

### PR DESCRIPTION
**Pull request by m-horky**
When HTTP_PROXY environment variable is set, and is pointing at valid
proxy server, these two tests would fail because insights-client reports
different errors on network errors for proxy/non-proxy scenarios.

This code takes proxies into account, and asserts strings accordingly.

---

- `el9` (all of RHEL 9)
- `el8` (all of RHEL 8)

* Card ID: CCT-1199

### CODE REVIEW

File: integration-tests/test_connection.py [hunk 1]
```diff
@@ -6,13 +6,31 @@
...
+
 # the tests need a valid connection to insights so therefore the subman registration
 pytestmark = pytest.mark.usefixtures("register_subman")
 
 
-def test_connection(insights_client):
+def _is_using_proxy(
+    config: "pytest_client_tools.insights_client.InsightsClientConfig",
+) -> bool:
+    if os.getenv("HTTP_PROXY") is not None:
```
Comment (ptoscano) [inline integration-tests/test_connection.py:24]: eh, I was setting a proxy in Unix would be this easy... I think it should be better to also check for `http_proxy` (lowercase) which (according to my experience) should be more commonly used

... and actually insights-core does not use `HTTP_PROXY` (which explicitly rejects), but `HTTPS_PROXY`:

https://github.com/RedHatInsights/insights-core/blob/23157b137ffa6ca3ee9f0d12ecf13e23f5f4a80d/insights/client/config.py#L553-L561

also, please keep into account that the proxy can be configured also in `rhsm.conf`, in case of autoconfig (which is true by default), so most likely you need to check that too, for added fun...

https://github.com/RedHatInsights/insights-core/blob/23157b137ffa6ca3ee9f0d12ecf13e23f5f4a80d/insights/client/connection.py#L334-L339

**Review by ptoscano — CHANGES_REQUESTED** [review]

File: integration-tests/test_connection.py [hunk 2]
```diff
@@ -6,12 +6,43 @@
...
+    for key in os.environ.keys():
+        if key.lower() == "https_proxy":
+            return True
+
+    try:
+        config_proxy = insights_config.proxy
+        if config_proxy != "":
+            return True
+    except KeyError:
+        return False
```
Comment (ptoscano) [inline integration-tests/test_connection.py:34]: shouldn't this be a simple `pass` (or `contextlib.suppress()`)? otherwise a missing `proxy` entry in `insights-client.conf` (which is the default) will cause `KeyError` to be raised, and thus `False` to be returned unconditionally, without trying to read `rhsm.conf`

File: integration-tests/test_connection.py [hunk 3]
```diff
@@ -6,12 +6,43 @@
...
+
+    # sub-man fixture doesn't currently support reading the configuration file,
+    # let's parse it ourselves.
+    rhsm_config = configparser.ConfigParser()
+    rhsm_config.read("/etc/rhsm/rhsm.conf")
+    if rhsm_config.get("server", "proxy_hostname", fallback=None) is not None:
+        return True
```
Comment (ptoscano) [inline integration-tests/test_connection.py:41]: even if it's a very narrow corner case, I'd catch all the exceptions of `ConfigParser` and ignore them, to avoid stacktracing on an invalid `rhsm.conf` or in case the `server` section or the `proxy_hostname` key are missing

**Review by ptoscano — CHANGES_REQUESTED** [review]
thanks for the improvements! there are a couple of changes needed to make this new code a bit safer

**Review by ptoscano — APPROVED** [review]
thanks!
